src/hooks/useSavedLayout.js

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from '@bpmn-io/properties-panel/preact/hooks';
+import { merge } from 'min-dash';
+
+
+/**
+ * Reacts to changes in the layout and saves the current state in local storage.
+ * Merges the saved layout with the layoutConfig, where layoutConfig has precedence.
+ *
+ * @param {Object} layoutConfig
+ * @param {EventBus} eventBus
+ * @returns {Object} layout
+ */
+export function useSavedLayout(
+    layoutConfig,
+    eventBus // eventbus must be injected because this hook is used outside of the BpmnPropertiesPanelContext
+) {
+
+  // Save layout changes
+  const [ layout, setLayout ] = useState(merge(
+    {},
+    getSavedLayout(),
+    layoutConfig
+  ));
+
+  useEffect(() => {
+    const cb = (e) => {
+      const { layout, skipApply } = e;
+
+      saveLayout(layout);
+
+      if (skipApply) {
+        return;
+      }
+
+      setLayout(layout);
+    };
+
+    eventBus.on('propertiesPanel.layoutChanged', cb);
+    return () => eventBus.off('propertiesPanel.layoutChanged', cb);
+  }, [ eventBus ]);
+
+
+  return layout;
+}
+
+// helpers //////
+function getSavedLayout() {
+  const savedLayout = window.localStorage.getItem('bpmn-properties-panel-layout') || '{}';
```

Comment:
I'm not sure if we want to hard code this inside the properties panel. 

I'd rather

* Provide a clear api / contract to listen for properties panel layout changes
* Instantiate the properties panel or update it with a shared layout

:arrow_right: Be API first, and not bury this inside the properties panel. How saving happens (if remotely or locally) is an application concern I think.

Reply:
If we provide such a hook then could we make saving and getting configurable to accomplish https://github.com/bpmn-io/bpmn-js-properties-panel/pull/891/files#r1140288946?

Reply:
I agree that the API should be explicit. I added [a172bd0](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/891/commits/a172bd038031babb9d6fed7f35f7c73d6fdbb70b) to make the API more explicit and also expose it as `propertiesPanel.setLayout()`.

I disagree that the implementation should always be on the integration side. In my opinion, it makes sense to have sensible defaults in the core which can be overwritten by the application. This is the case here: you can implement your own persistence/update logic by subscribing to `propertiesPanel.layoutChanged` and using the API (either eventBus or `propertiesPanel.setLayout`). 
If you don't implement your own handler, you still get basic persistence across tabs in localStorage as the default implementation.

Reply:
Let's say that this would be the first time that we move persistent state inside our libraries :).

Reply:
Fair point. Removed with [c82d5b3](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/891/commits/c82d5b376084c99a3a7d461c7a7458aaa21e1f1e)